conn.go

```diff
@@ -981,7 +1002,9 @@ func (c *Conn) handleProtocolError(message string) error {
 	if len(data) > maxControlFramePayloadSize {
 		data = data[:maxControlFramePayloadSize]
 	}
-	c.WriteControl(CloseMessage, data, time.Now().Add(writeWait))
+	if err := c.WriteControl(CloseMessage, data, time.Now().Add(writeWait)); err != nil {
+		return err
```

Comment:
The close message error is secondary to the protocol error.  Revert to the original or use `errors.Join( errors.New("websocket: " + message), c.WriteControl(CloseMessage, data, time.Now().Add(writeWait)))`.